backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/salesrep/process/CreateMissingCommissionEntriesForSalesRep.java

```diff
@@ -0,0 +1,432 @@
+		final Iterator<ChangeLogEntry> salesRepLogIterator = orderedChangeLogEntries.iterator();
+
+		while (salesRepLogIterator.hasNext())
+		{
+			final ChangeLogEntry logEntry = salesRepLogIterator.next();
+
+			// we only care about the newValue if it is the last entry in the change log
+			if (!salesRepLogIterator.hasNext())
+			{
+				if (logEntry.getOldValue() != null)
+				{
+					final InstantInterval oldValueInterval = InstantInterval.of(currentFrom, logEntry.getChangeTimestamp());
+
+					valueByInterval.put(oldValueInterval, logEntry.getOldValue());
+				}
+
+				if (logEntry.getNewValue() != null)
+				{
+					final InstantInterval newValueInterval = InstantInterval.of(logEntry.getChangeTimestamp(), targetInterval.getTo());
+
+					valueByInterval.put(newValueInterval, logEntry.getNewValue());
+				}
+			}
+			else if (logEntry.getOldValue() != null)
+			{
+				final InstantInterval oldValueInterval = InstantInterval.of(currentFrom, logEntry.getChangeTimestamp());
+
+				valueByInterval.put(oldValueInterval, logEntry.getOldValue());
+			}
+
+			currentFrom = logEntry.getChangeTimestamp();
+		}
+
+		return ImmutableMap.copyOf(valueByInterval);
+	}
+
+	private Optional<BPartnerId> getSuperSalesRepBPId(@NonNull final String superSalesRepUserIdValue, @NonNull final BPartnerId salesRepId)
+	{
+		final int userId = NumberUtils.asInt(superSalesRepUserIdValue, -1);
+		final UserId superSalesRepUserId = UserId.ofRepoIdOrNull(userId);
+
+		if (superSalesRepUserId == null)
+		{
+			Loggables.withLogger(logger, Level.WARN)
+					.addLog("*** WARN during calculation for bPartnerId: {}:"
+							+ " salesRepUser value: {} cannot be converted to int! Skipping.. ", superSalesRepUserIdValue);
+			return Optional.empty();
+		}
+
+		final I_AD_User superSalesRepUser = userDAO.getById(superSalesRepUserId);
+
+		if (superSalesRepUser.getC_BPartner_ID() <= 0)
+		{
+			Loggables.withLogger(logger, Level.WARN)
+					.addLog("*** WARN during calculation for bPartnerId: {}:"
+							+ " salesRepUser: {} doesn't have a bPartner! Skipping.. ", salesRepId, superSalesRepUser.getAD_User_ID());
+			return Optional.empty();
+		}
+		else
+		{
+			final I_C_BPartner superSalesRepBPartner = bPartnerDAO.getById(superSalesRepUser.getC_BPartner_ID());
+
+			if (!superSalesRepBPartner.isSalesRep())
+			{
+				Loggables.withLogger(logger, Level.WARN)
+						.addLog("*** WARN during calculation for bPartnerId: {}: missing 'isSalesRep' flag on regional manager: {}! Skipping.. ",
+								salesRepId, superSalesRepBPartner.getC_BPartner_ID());
+
+				return Optional.empty();
+			}
+			else
+			{
+				return Optional.of(BPartnerId.ofRepoId(superSalesRepBPartner.getC_BPartner_ID()));
+			}
+		}
+	}
+
+	public void recalculateCommissionForCriteria(@NonNull final RecalculateCommissionCriteria commissionCriteria)
+	{
+		bPartnerBL.setBPartnerSalesRepId(commissionCriteria.getBPartnerId(), commissionCriteria.getTopLevelSalesRepId()); // set within the current transaction
```

Comment:
problem here: it's sett in the current trx, but when we read this value in `CommissionHierarchyFactory#createFor()`, 
we read ot from a C_BPartner that was loaded with InterfaceWrapperHelper.loadOutOfTrx()
So there in CommissionHierarchyFactory, the C_BPartner does not have the TopLevelSalesRepId we are setting here and therefor we get no hierarchy

Suggestion 1: change CommissionHierarchyFactory to load it with `InterfaceWrapperHelper.load` (within-trx), and remove the new trxManager.runInNewTrx from line 339 (otherwise "within-trx" is within *another* trx)
Suggestion 2: change IBPartnerBL.setBPartnerSalesRepId to same the changed record out of trx as well